### DISCUSSION TITLE
Build TypeError when using scope hoist

### DISCUSSION THREAD

**Discussion by laurentlucian** [parcel-bundler/parcel#4753] [General] [answered]
I saw some recent issues with scope hoist, and I'm running into a similar error.
I'm not sure if it's a parcel bug yet so I'm creating a discussion.

I'm using `"parcel": "^2.0.0-nightly.296"`

I get a blank screen in production build with the following error (no errors while building)
![image](https://user-images.githubusercontent.com/33508364/84959144-a27da800-b0b3-11ea-9bcf-ac37a7e1088e.png)
However, building it without scope hoisting runs perfectly fine.

Can anyone help?

**Comment by mischnic** [comment, marked as answer]
Sounds like a bug. Can you share a code sample so that I can reproduce the error?

**Reply by laurentlucian** [reply]
I narrowed it down to Chakra `CSSReset` or `ThemeProvider` with .babelrc env and react presets
https://github.com/laurentlucian/parcel-hoist-css

**Reply by mischnic** [reply]
The problem is that using plain `@babel/preset-env` also transpiles ESM to CommonJS which makes our scope hoisting really ineffective (and in this case breaks it as well)

`@parcel/babel-preset-env` is a wrapper around `@babel/preset-env` that makes sure Parcel's defaults are forwarded (and it should work with other tools as well, e.g. test runners).
```js
{
	"presets": ["@parcel/babel-preset-env",  "@babel/preset-react"]
}
```

Alternatively, you could do this:

```
{
	"presets": [
		[
			"@babel/preset-env",
			{
				"modules": false
			}
		],
		"@babel/preset-react"
	]
}
```
